### PR TITLE
If legacy MAP, use default poster image instead of image from page data

### LINKED ISSUE #5591: Ensure TC2 Asset Poster Image is visible when links are shared
**Is your feature request related to a problem? Please describe.**
We want to ensure that the poster image is added to the og:image 
Currently when you share an pal TC2 asset link on social you get a preview of a the poster image
We want to ensure the same behaviour applies to TC2 links on Simorgh. 

**Describe the solution you'd like**
Set the og:image as the poster image
if a poster image isn't available it should default to the service logo. 

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]
Can be tested using the structured data testing tool: https://developers.facebook.com/tools/debug/

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves #5591

**Overall change:**
- Check if the asset is a legacy MAP, and if so, set the image locator to null. Downstream, when this is used to populate the `og:image` metadata tag, a null image locator means that the default image will be used e.g. https://news.files.bbci.co.uk/ws/img/logos/og/arabic.png

The way in which we have configured ichef + image branding to work with CPS images, would require a large set of code changes to use the image index from the page data, which is not really feasible for TC2 MAP pages, given that traffic is low & content is rarely shared.

**Code changes:**
- if legacy MAP, return null as image locator
- Updating snapshots for integration tests

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/pages/MediaAssetPage/MediaAssetPage.jsx [hunk 1]
```diff
@@ -53,9 +53,18 @@ const MediaAssetPage = ({ pageData }) => {
...
   );
   const indexImagePath = path(['promo', 'indexImage', 'path'], pageData);
-  const indexImageLocator = indexImagePath
-    ? getImageParts(indexImagePath)[1]
-    : null;
+
+  const getIndexImageLocator = () => {
+    const imageParts = indexImagePath && getImageParts(indexImagePath);
+
+    if (isLegacyMediaAssetPage(requestContext.canonicalLink)) {
```
Comment (RichardPK) [inline src/app/pages/MediaAssetPage/MediaAssetPage.jsx]: Could bump this to line 58 so it gets hit first in the function & only does the getImageParts logic if it isn't a legacy asset? Marginal improvement 🤷‍♂️

**Review by RichardPK — APPROVED** [review]

File: src/app/pages/MediaAssetPage/MediaAssetPage.jsx (hunk 1 above)
Comment (karinathomasbbc) [reply src/app/pages/MediaAssetPage/MediaAssetPage.jsx]: Good shout, will do that

File: src/app/pages/MediaAssetPage/MediaAssetPage.jsx [hunk 2]
```diff
@@ -52,10 +52,18 @@ const MediaAssetPage = ({ pageData }) => {
...
-    ? getImageParts(indexImagePath)[1]
-    : null;
+
+  const getIndexImageLocator = () => {
+    if (isLegacyMediaAssetPage(requestContext.canonicalLink)) {
+      return null;
+    }
+
+    const indexImagePath = path(['promo', 'indexImage', 'path'], pageData);
+    const imageParts = indexImagePath && getImageParts(indexImagePath);
```
Comment (rebeccamcginn) [inline src/app/pages/MediaAssetPage/MediaAssetPage.jsx]: This is quite hard to read. If we check that indexImagePath exists before line 62, we can get rid of the checks on lines 62 & 63

Comment (karinathomasbbc) [reply src/app/pages/MediaAssetPage/MediaAssetPage.jsx]: Could you suggest some pseudocode which would make it easier to read?

Comment (rebeccamcginn) [reply src/app/pages/MediaAssetPage/MediaAssetPage.jsx]: Check if indexImagePath has a value. If it doesn't, return early. Then remove the `indexImagePath &&` and `imageParts &&` checks as they will no longer be needed

File: src/app/pages/MediaAssetPage/MediaAssetPage.jsx [hunk 3]
```diff
@@ -52,10 +52,26 @@ const MediaAssetPage = ({ pageData }) => {
...
+
+      if (imageParts && imageParts.length >= 2) {
+        return imageParts[1];
+      }
+    }
+
+    return null;
+  };
+
+  const indexImageLocator = getIndexImageLocator();
```
Comment (ryanmccombe) [inline src/app/pages/MediaAssetPage/MediaAssetPage.jsx]: Ramda could simplify this a lot, something like:

```javascript
const getIndexImageLocator = pipe(
  pathOr('', ['promo', 'indexImage', 'path']),
  getImageParts,
  last,
);

const indexImageLocator = isLegacyMediaAssetPage ? null : getIndexImageLocator(pageData);
```

Comment (karinathomasbbc) [reply src/app/pages/MediaAssetPage/MediaAssetPage.jsx]: I'll give it a shot to see what happens 😬

Comment (karinathomasbbc) [reply src/app/pages/MediaAssetPage/MediaAssetPage.jsx]: `pipe` didn't quite work - but I used `last` instead. I think it's _slightly_ easier to read now?

**Review by ryanmccombe — APPROVED** [review]
👍

**Pull request merged**